### PR TITLE
checker: when there are too many orphan peers, try to delete them

### PR DESCRIPTION
**Pull request by HunDunDM**
Signed-off-by: HunDunDM <hundundm@gmail.com>

### What problem does this PR solve?

Issue Number: Ref #5786 Part.1

### What is changed and how does it work?

checker: when there are too many orphan peers, try to delete them

### Check List

Tests

- Unit test

Code changes

Side effects

Related changes

- Need to cherry-pick to the release branch

### Release note

```release-note
Fix an issue that PD may repeatedly add Learner to a Region.
```

### CODE REVIEW

**Comment by nolouch** [comment]
@HunDunDM  why hold?

**Review by nolouch — APPROVED** [review]
lgtm

**Comment by nolouch** [comment]
cc @lhy1024 @rleungx

File: server/schedule/checker/rule_checker.go [hunk 1]
```diff
@@ -390,31 +390,52 @@ func (c *RuleChecker) fixOrphanPeers(region *core.RegionInfo, fit *placement.Reg
...
+			if downPeer.Peer.GetId() == id {
+				return true
+			}
+		}
+		return false
+	}
 	// remove orphan peers only when all rules are satisfied (count+role) and all peers selected
 	// by RuleFits is not pending or down.
+	hasUnhealthyFit := false
+loopFits:
```
Comment (lhy1024) [inline server/schedule/checker/rule_checker.go:409]: why to add loop?

**Review by rleungx — APPROVED** [review]

**Pull request merged**